frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/FetchBackgroundable.java

```diff
@@ -50,14 +53,18 @@ public void run(ProgressIndicator indicator) {
       return;
     }
     var fetchResult = fetchSupport.fetch(gitRepository, remote, refspec);
-    fetchResult.showNotificationIfFailed(
-        format(getString("action.GitMachete.FetchBackgroundable.notification.title.fetch-fail"), refspec));
+    fetchResult.showNotificationIfFailed(failureNotificationText);
+    fetchResult.throwExceptionIfFailed();
```

Comment:
Hmm so what happened before this line has been added?

Reply:
both success and failure notifications have been shown